qurator/eynollah/utils/drop_capitals.py

```diff
@@ -142,53 +142,54 @@ def adhere_drop_capital_region_into_corresponding_textline(
+                    img_textlines = cv2.fillPoly(img_textlines, pts=[polygons_of_drop_capitals[i_drop]], color=(255, 255, 255))
 
-                img_textlines = img_textlines.astype(np.uint8)
+                    img_textlines = img_textlines.astype(np.uint8)
 
-                # plt.imshow(img_textlines)
-                # plt.show()
-                imgray = cv2.cvtColor(img_textlines, cv2.COLOR_BGR2GRAY)
-                ret, thresh = cv2.threshold(imgray, 0, 255, 0)
+                    # plt.imshow(img_textlines)
+                    # plt.show()
+                    imgray = cv2.cvtColor(img_textlines, cv2.COLOR_BGR2GRAY)
+                    ret, thresh = cv2.threshold(imgray, 0, 255, 0)
 
-                contours_combined, hierarchy = cv2.findContours(thresh, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+                    contours_combined, hierarchy = cv2.findContours(thresh, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
 
-                # print(len(contours_combined),'len textlines mixed')
-                areas_cnt_text = np.array([cv2.contourArea(contours_combined[j]) for j in range(len(contours_combined))])
+                    # print(len(contours_combined),'len textlines mixed')
+                    areas_cnt_text = np.array([cv2.contourArea(contours_combined[j]) for j in range(len(contours_combined))])
 
-                contours_biggest = contours_combined[np.argmax(areas_cnt_text)]
+                    contours_biggest = contours_combined[np.argmax(areas_cnt_text)]
 
-                # print(np.shape(contours_biggest))
-                # print(contours_biggest[:])
-                # contours_biggest[:,0,0]=contours_biggest[:,0,0]#-all_box_coord[int(region_final)][2]
-                # contours_biggest[:,0,1]=contours_biggest[:,0,1]#-all_box_coord[int(region_final)][0]
-                # print(np.shape(contours_biggest),'contours_biggest')
-                # print(np.shape(all_found_texline_polygons[int(region_final)][arg_min]))
-                ##contours_biggest=contours_biggest.reshape(np.shape(contours_biggest)[0],np.shape(contours_biggest)[2])
-                all_found_texline_polygons[int(region_final)][arg_min] = contours_biggest
+                    # print(np.shape(contours_biggest))
+                    # print(contours_biggest[:])
+                    # contours_biggest[:,0,0]=contours_biggest[:,0,0]#-all_box_coord[int(region_final)][2]
+                    # contours_biggest[:,0,1]=contours_biggest[:,0,1]#-all_box_coord[int(region_final)][0]
+                    # print(np.shape(contours_biggest),'contours_biggest')
+                    # print(np.shape(all_found_texline_polygons[int(region_final)][arg_min]))
+                    ##contours_biggest=contours_biggest.reshape(np.shape(contours_biggest)[0],np.shape(contours_biggest)[2])
+                    all_found_texline_polygons[int(region_final)][arg_min] = contours_biggest
+                except:
```

Comment:
This fixes the issue in the short-term but it might hide problems in the future. Can you specify what kind of Exception could happen here and only catch that? And ideally keep the statements in the `try` clause as short as possible? It's more effort, true, but will help maintainenance in the long term.